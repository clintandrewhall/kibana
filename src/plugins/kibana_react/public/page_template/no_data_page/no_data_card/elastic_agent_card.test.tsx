/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { shallow } from 'enzyme';
import React from 'react';
import { NoDataElasticAgentCard } from './elastic_agent_card';

jest.mock('../../../context', () => ({
  ...jest.requireActual('../../../context'),
  useKibana: jest.fn().mockReturnValue({
    services: {
      http: {
        basePath: { prepend: jest.fn((path: string) => (path ? path : 'path')) },
      },
      application: { capabilities: { navLinks: { integrations: true } } },
      uiSettings: { get: jest.fn() },
    },
  }),
}));

describe('NoDataElasticAgentCard', () => {
  test('renders', () => {
    const component = shallow(<NoDataElasticAgentCard solution="Solution" />);
    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    test('recommended', () => {
      const component = shallow(<NoDataElasticAgentCard recommended solution="Solution" />);
      expect(component).toMatchSnapshot();
    });

    test('button', () => {
      const component = shallow(<NoDataElasticAgentCard button="Button" solution="Solution" />);
      expect(component).toMatchSnapshot();
    });

    test('href', () => {
      const component = shallow(
        <NoDataElasticAgentCard href="#" button="Button" solution="Solution" />
      );
      expect(component).toMatchSnapshot();
    });

    test('category', () => {
      const component = shallow(<NoDataElasticAgentCard category="custom" solution="Solution" />);
      expect(component).toMatchSnapshot();
    });
  });
});
