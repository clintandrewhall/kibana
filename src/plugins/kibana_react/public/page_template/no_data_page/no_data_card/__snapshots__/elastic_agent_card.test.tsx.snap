// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoDataElasticAgentCard props button 1`] = `
<RedirectAppLinks
  application={
    Object {
      "capabilities": Object {
        "navLinks": Object {
          "integrations": true,
        },
      },
    }
  }
>
  <EuiCard
    betaBadgeProps={
      Object {
        "label": undefined,
      }
    }
    description="Use Elastic Agent for a simple, unified way to collect data from your machines."
    footer={
      <EuiButton
        fill={true}
      >
        Button
      </EuiButton>
    }
    href="/app/integrations/browse"
    image={
      <EuiImage
        alt=""
        size="fullWidth"
        style={
          Object {
            "background": "aliceblue",
            "height": 240,
            "objectFit": "cover",
            "width": "max(100%, 360px)",
          }
        }
        url="/plugins/kibanaReact/assets/elastic_agent_card.svg"
      />
    }
    paddingSize="l"
    title={
      <EuiScreenReaderOnly>
        <span>
          Add Elastic Agent
        </span>
      </EuiScreenReaderOnly>
    }
  />
</RedirectAppLinks>
`;

exports[`NoDataElasticAgentCard props category 1`] = `
<RedirectAppLinks
  application={
    Object {
      "capabilities": Object {
        "navLinks": Object {
          "integrations": true,
        },
      },
    }
  }
>
  <EuiCard
    betaBadgeProps={
      Object {
        "label": undefined,
      }
    }
    description="Use Elastic Agent for a simple, unified way to collect data from your machines."
    footer={
      <EuiButton
        fill={true}
      >
        Add Elastic Agent
      </EuiButton>
    }
    href="/app/integrations/browse/custom"
    image={
      <EuiImage
        alt=""
        size="fullWidth"
        style={
          Object {
            "background": "aliceblue",
            "height": 240,
            "objectFit": "cover",
            "width": "max(100%, 360px)",
          }
        }
        url="/plugins/kibanaReact/assets/elastic_agent_card.svg"
      />
    }
    paddingSize="l"
    title={
      <EuiScreenReaderOnly>
        <span>
          Add Elastic Agent
        </span>
      </EuiScreenReaderOnly>
    }
  />
</RedirectAppLinks>
`;

exports[`NoDataElasticAgentCard props href 1`] = `
<RedirectAppLinks
  application={
    Object {
      "capabilities": Object {
        "navLinks": Object {
          "integrations": true,
        },
      },
    }
  }
>
  <EuiCard
    betaBadgeProps={
      Object {
        "label": undefined,
      }
    }
    description="Use Elastic Agent for a simple, unified way to collect data from your machines."
    footer={
      <EuiButton
        fill={true}
      >
        Button
      </EuiButton>
    }
    href="#"
    image={
      <EuiImage
        alt=""
        size="fullWidth"
        style={
          Object {
            "background": "aliceblue",
            "height": 240,
            "objectFit": "cover",
            "width": "max(100%, 360px)",
          }
        }
        url="/plugins/kibanaReact/assets/elastic_agent_card.svg"
      />
    }
    paddingSize="l"
    title={
      <EuiScreenReaderOnly>
        <span>
          Add Elastic Agent
        </span>
      </EuiScreenReaderOnly>
    }
  />
</RedirectAppLinks>
`;

exports[`NoDataElasticAgentCard props recommended 1`] = `
<RedirectAppLinks
  application={
    Object {
      "capabilities": Object {
        "navLinks": Object {
          "integrations": true,
        },
      },
    }
  }
>
  <EuiCard
    betaBadgeProps={
      Object {
        "label": "Recommended",
      }
    }
    description="Use Elastic Agent for a simple, unified way to collect data from your machines."
    footer={
      <EuiButton
        fill={true}
      >
        Add Elastic Agent
      </EuiButton>
    }
    href="/app/integrations/browse"
    image={
      <EuiImage
        alt=""
        size="fullWidth"
        style={
          Object {
            "background": "aliceblue",
            "height": 240,
            "objectFit": "cover",
            "width": "max(100%, 360px)",
          }
        }
        url="/plugins/kibanaReact/assets/elastic_agent_card.svg"
      />
    }
    paddingSize="l"
    title={
      <EuiScreenReaderOnly>
        <span>
          Add Elastic Agent
        </span>
      </EuiScreenReaderOnly>
    }
  />
</RedirectAppLinks>
`;

exports[`NoDataElasticAgentCard renders 1`] = `
<RedirectAppLinks
  application={
    Object {
      "capabilities": Object {
        "navLinks": Object {
          "integrations": true,
        },
      },
    }
  }
>
  <EuiCard
    betaBadgeProps={
      Object {
        "label": undefined,
      }
    }
    description="Use Elastic Agent for a simple, unified way to collect data from your machines."
    footer={
      <EuiButton
        fill={true}
      >
        Add Elastic Agent
      </EuiButton>
    }
    href="/app/integrations/browse"
    image={
      <EuiImage
        alt=""
        size="fullWidth"
        style={
          Object {
            "background": "aliceblue",
            "height": 240,
            "objectFit": "cover",
            "width": "max(100%, 360px)",
          }
        }
        url="/plugins/kibanaReact/assets/elastic_agent_card.svg"
      />
    }
    paddingSize="l"
    title={
      <EuiScreenReaderOnly>
        <span>
          Add Elastic Agent
        </span>
      </EuiScreenReaderOnly>
    }
  />
</RedirectAppLinks>
`;
